### ISSUE TITLE
[ ERROR ] Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.front.ChangePlaceholderTypes.ChangePlaceholderTypes'>): Something bad has happened with graph! Data node "Preprocessor/mul" has 2 producers

### ISSUE THREAD

**Issue by NatachaRuchaudDB** [openvinotoolkit/openvino#2611]
- OpenVINO => 2020.4
- Operating System / Platform => Windows 64 Bit
- Problem classification: Model Conversion
- Framework: TensorFlow 1.15
- Model name: ssdlite_mobilenet_v2_coco_2018_05_09
- Object detection tensorflow api: master

##### Detailed description of the problem

Hello,
I retrained the ssdlite_mobilenet_v2_coco_2018_05_09 model with my own data on the last version of object detection tensorflow api and also with older version such as v1.13.0 but I cannot convert the model to Openvino. 
I have the following error:
[ ERROR ] Exception occurred during running replacer "REPLACEMENT_ID" (<class 'extensions.front.ChangePlaceholderTypes.ChangePlaceholderTypes'>): Something bad has happened with graph! Data node "Preprocessor/mul" has 2 producers
I tried many version of object detection tensorflow api, tensorflow and openvino but each time I have the same error...

However, I succeed to convert the model of ssdlite_mobilenet_v2_coco_2018_05_09 provided by the zoo library.

Please if you have any suggestion.

**Comment by jgespino** [comment]
Hi @NatachaRuchaudDB 

Could you please provide the model optimizer command you are using? Please ensure you are using the configuration file for TensorFlow Object Detection API with `--transformations_config`, see [Converting TensorFlow Object Detection API Models](https://docs.openvinotoolkit.org/latest/openvino_docs_MO_DG_prepare_model_convert_model_tf_specific_Convert_Object_Detection_API_Models.html#how_to_convert_a_model). I also recommend upgrading to the latest OpenVINO 2021.1 release.

Regards,
Jesus

**Comment by NatachaRuchaudDB** [comment]
Hi @jgespino 

The model optimizer command that I use is:

python mo_tf.py  --input_model ..\modelTrain\frozen_inference_graph.pb  --output_dir ..\modelTrain --transformations_config extensions/front/tf/ssd_support_api_v1.15.json  --tensorflow_object_detection_api_pipeline_config ..\modelTrain\pipeline.config

When I use this command to convert the same model downloaded from the TensorFlow Object Detection API everything is ok. I convert perfectly the tensorflow model to OpenVino.
It is when I re train this model with the TensorFlow Object Detection API I do not succeed to convert to OpenVino the new trained model which is exactly the same than the one of downloading from the TensorFlow Object Detection API (same topology etc..).

I already tried many version of the TensorFlow Object Detection API / Tensorflow / OpenVino with no success. I will re try with OpenVino 2021.1 release and I let you know. 

Thank you for your response.
Best,
Natacha.

**Comment by jgespino** [comment]
Hi @NatachaRuchaudDB ,

Looks like you are using the model optimizer correctly, we had a previous report of a similar issue on our Intel Community. User reported the issue being solved by using a TensorFlow branch prior to July 1st. Could you give that a try?

See discussion here:
https://community.intel.com/t5/Intel-Distribution-of-OpenVINO/TF-ssdv2-to-IRv10-error-quot-Preprocessor-mul-quot-has-2/td-p/1189093

Regards,
Jesus

**Comment by NatachaRuchaudDB** [comment]
Hi @jgespino  

Yes, I also trained the model using a prior version to July 1st of TF Obj Det API and even other prior versions (2 years ago). Unfortunately, I get the same error, so it is very strange.

Best,
Natacha.

**Comment by jgespino** [comment]
Hi @NatachaRuchaudDB 

I apologies for the delay for some reason I missed the notification. Could you provide your TensorFlow model?

Regards,
Jesus

**Comment by NatachaRuchaudDB** [comment]
Hi @jgespino 

No problem. Yes I can share the model. Where do you want that I send the Tensorflow model ?

Best,
Natacha.

**Comment by jgespino** [comment]
Hi @NatachaRuchaudDB 

You can share it here or send it to my email if you prefer to share it privately.

Regards,
Jesus

**Comment by jgespino** [comment]
Hi @NatachaRuchaudDB 

Thanks for sending over the model, I took a look and it seems there are changes to the order of applying input image normalization and resizing images. The dev team is already looking into this issue, however, I don't have an ETA on fix yet.

Regards,
Jesus

Ref. 37922

**Comment by NatachaRuchaudDB** [comment]
Hi @jgespino 

Thanks you for having a look. I hope you will find a solution.

Best,
Natacha.

**Comment by lazarevevgeny** [comment]
Hi @NatachaRuchaudDB ,

please, use the MO from the https://github.com/openvinotoolkit/openvino/pull/3063 and let me know if the solution works for you.

**Comment by NatachaRuchaudDB** [comment]
Hi @lazarevevgeny 

Thanks ! I tested. The conversion to OpenVino works but unforthunately not the inferences ... I obtain no detections. The prediction vector is always [-1, 0, 0, 0, ...0, 0]. When I run inferences with Tensorflow I obtain mutliple detections. And when I use OpenVino with models not from the TF API I obtain detections.

Best,
Natacha.

**Comment by lazarevevgeny** [comment]
@NatachaRuchaudDB , could you share the sample image with @jgespino for the model you shared with him?

**Comment by NatachaRuchaudDB** [comment]
@lazarevevgeny 

Yes, sure. For the inferences, I use the test images of HollywoodHead database available on https://www.di.ens.fr/willow/research/headdetection/
I hope it answer to your question.

Best, 
Natacha.

**Comment by lazarevevgeny** [comment]
@NatachaRuchaudDB , I did the following using MO and IE from the master branch:

> ./mo.py --input_model frozen_inference_graph.pb --transformation extensions/front/tf/ssd_support_api_v1.15.json --tensorflow_object_detection pipeline.config  --reverse_input_channels

Pay attention to the "--reverse_input_channels" in the command line.

Then I used the object_detection_sample_ssd:

> ./object_detection_sample_ssd -m frozen_inference_graph.xml -i casablanca_074268.jpeg

And I see the detections:

![image](https://user-images.githubusercontent.com/30651952/99047191-e3b3ae80-25a4-11eb-86bf-e30a4a5c07b7.png)

**Comment by NatachaRuchaudDB** [comment]
Hello @lazarevevgeny 

Thanks a lot !! It works !!!

Best,
Natacha Ruchaud.

**Comment by lazarevevgeny** [comment]
> Hello @lazarevevgeny
> 
> Thanks a lot !! It works !!!
> 
> Best,
> Natacha Ruchaud.

Great to hear that! Closing the issue.
